requests/adapters.py

```diff
@@ -10,6 +10,10 @@
 
 import os.path
 import socket
+try:
+    from threading import RLock
+except ImportError:  # threading is an optional module and may not be present.
```

Comment:
It looks like this [was a problem](https://github.com/shazow/urllib3/issues/289) in urllib3 which is where I originally learned threading is optional. I used [dummy_threading](https://docs.python.org/3.6/library/dummy_threading.html#module-dummy_threading) since that's what the Python documentation recommended, although it looks like urllib3 just defines the required interface itself.